components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/GroupsApiServiceImpl.java

```diff
@@ -9,43 +13,91 @@
 
 import java.io.InputStream;
 
+import org.wso2.carbon.auth.scim.rest.api.util.ApiServiceUtils;
+import org.wso2.charon3.core.extensions.UserManager;
+import org.wso2.charon3.core.protocol.SCIMResponse;
+import org.wso2.charon3.core.protocol.endpoints.GroupResourceManager;
+import org.wso2.charon3.core.protocol.endpoints.UserResourceManager;
 import org.wso2.msf4j.formparam.FormDataParam;
 import org.wso2.msf4j.formparam.FileInfo;
 import org.wso2.msf4j.Request;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.SecurityContext;
 
 public class GroupsApiServiceImpl extends GroupsApiService {
+    private static final Log LOG = LogFactory.getLog(GroupsApiServiceImpl.class);
+
     @Override
     public Response groupsGet(Integer startIndex
 , Integer count
 , String filter
   ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            GroupResourceManager groupResourceManager = new GroupResourceManager();
+            SCIMResponse scimResponse = groupResourceManager.listWithGET(userManager,  filter, startIndex, count,
+                    null, null, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
```

Comment:
Exception has not been logged. Need to log exception as follows,

log.error("Message", e);